package-scope.qmd

```diff
@@ -0,0 +1,55 @@
+---
+title: "Package scope"
+---
+
+## Visualisation in R packages
+
+The use of visualisation is a fundamental step in scientific analysis and a [core component of data science](https://r4ds.had.co.nz/introduction.html). However, the question of whether a package should include plotting functionality is not obvious.
+
+::: {.callout-note title="Not all package plotting requires are equal"}
+
+This section details the Epiverse-TRACE reasoning for the inclusion or exclusion of functionality around plotting. It is not a commentary generalisation to all R packages.
+
+:::
+
+## Epiverse-TRACE philosophy
+
+The Epiverse-TRACE philosophy is to only include minimal, dependency-free, plotting functionality, and to instead provide examples and templates of plotting using {ggplot2} in vignettes. An example of the inclusion of a plotting function is the [`<epidist>` plotting method](https://epiverse-trace.github.io/epiparameter/reference/plot.epidist.html) in {epiparameter}.
+
+Epiverse-TRACE packages aim to provide comprehensive documentation that explains exported functions and demonstrates their applications, both through [{roxygen2}](https://roxygen2.r-lib.org/) function-specific documentation, or via longer-form vignettes. It is in the vignettes that Epiverse-TRACE packages show examples of how the data or model outputs can be visualised, with the associated code chunks required to produce the plot.
+
+It is with this provision of plotting code chunks we hope to provide the community with a resource from which their desired plots can be produced. Vignettes use the {ggplot2} package to provide easily extensible plots with the addition of [layers](https://rpubs.com/hadley/ggplot2-layers).
+
+Epiverse-TRACE packages are not developed uniformly and certain packages may deviate from this philosophy. In these cases the reasoning for the difference should be documented in that package's design vignette.
+
+## Epiverse-TRACE reasoning
+
+A benefit of including plotting functions is it enables package users to quickly inspect a feature of the package (e.g. model output) without having to write their own script or function. Despite this clear benefit of shipping plotting with a package there are several
+drawbacks, and here we outline the reasons why only dependency-free plotting (also commonly referred to as base R plotting) is included in Epiverse-TRACE R packages.
```

Comment:
I'm not completely sure we should frame this discussion as base vs ggplot. The discussion we had on slack didn't really make this distinction.

My overall take is that the consensus is to include no, or very basic plotting functions.

I would have similar concerns as what you mentioned for complex base plots (i.e., it's probably simpler and more flexible for users to have an example of how to do it, rather than an unflexible solution which may not fit their specific needs and design preferences)